src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+		{
+			for (GKInstance inst : cosmicObjects)
+			{
+				String identifier = (String)inst.getAttributeValue(ReactomeJavaConstants.identifier);
+			
+				COSMICIdentifierUpdater updater = new COSMICIdentifierUpdater();
+				updater.setIdentifier(identifier);
+				updater.setDbID(inst.getDBID());
+
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> EWASes = inst.getReferers(ReactomeJavaConstants.crossReference);
+				// If NO EWASes exist, then log this information.
+				if (EWASes == null || EWASes.isEmpty())
+				{
+					identifiersWithNoReferrerPrinter.printRecord(identifier);
+				}
+				else
+				{
+					// Check the EWASes for mismatches between referenceSequence identifier and main identifier.
+					// If there's a mismatch then suggest COSF, else suggest COSM.
+					checkEWASes(nonEWASPrinter, identifier, updater, EWASes);
+				}
+				
+				// If the identifier starts with C it's not a numeric identifier.
+				if (identifier.startsWith("C"))
+				{
+					// Consider: is it possible that we might get 1:n mapping for some identifier?
+					updates.put(updater.getIdentifier(), updater);
+				}
+				else
+				{
+					updates.put(updater.getSuggestedPrefix() + updater.getIdentifier(), updater);
+				}
+			}
+		}
+		return updates;
+	}
+
+	/**
+	 * Checks EWASes to see if they have modifiedResidues that have a referenceSequence that is NOT the same as the EWASes referenceEntity.
+	 * The suggested prefix will be set to COSF on the update record if mismatches are found, otherwise COSM will be set.
+	 * @param nonEWASPrinter CSVPrinter for reporting.
+	 * @param identifier Identifier of the object being checked, used for reporting.
+	 * @param updater A COSMICIdentifierUpdater whose suggested prefix will be updated.
+	 * @param EWASes The EWASes to check.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	private static void checkEWASes(CSVPrinter nonEWASPrinter, String identifier, COSMICIdentifierUpdater updater, Collection<GKInstance> EWASes) throws InvalidAttributeException, Exception, IOException
+	{
+		String prefix;
+		for (GKInstance ewas : EWASes)
+		{
+			if (ewas.getSchemClass().getName().equals(ReactomeJavaConstants.EntityWithAccessionedSequence))
+			{
+				GKInstance refSequence = (GKInstance) ewas.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+				// get hasModifiedResidue
+				@SuppressWarnings("unchecked")
+				List<GKInstance> modResidues = (List<GKInstance>) ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+				
+				boolean foundMismatchedRefSequence = referenceSequenceMismatchesResidues(refSequence, modResidues);
+				
+				// If we get to the end of the loop and there is a mismatch, then COSF.
+				prefix = foundMismatchedRefSequence ? COSMIC_FUSION_PREFIX : COSMIC_LEGACY_PREFIX;
+				updater.setSuggestedPrefix(prefix);
+			}
+			else
+			{
+				nonEWASPrinter.printRecord(identifier, ewas.toString());
```

Comment:
Won't printing a GKInstance to string also include the square brackets? Nothing wrong with that, but I'm not sure how the file produced is used and if it is anything consumed by users I think we tend to avoid the full instance name

Reply:
Here's an example of what it looks like:
```
[CandidateSet:1602342] FBXW7 R505/387C mutant [nucleoplasm]
```
The file is a CSV file. Lisa has already seen this in the report and has not had any problems with it. The purpose of this report is to let the curators know that there are non-EWAS objects identified with COSMIC identifiers.

Reply:
Ah OK, I figured it was internal use but I wasn't sure. Sounds good!